src/cloud/functions/leaderboards/around-me/leaderboard/run.csx

```diff
@@ -0,0 +1,160 @@
+        client = new DocumentClient(new Uri(endpoint), key);
+
+        // Create Database and Collection in Cosmos DB Account if they don't exist
+        await client.CreateDatabaseIfNotExistsAsync(new Database { Id = db });
+        await client.CreateDocumentCollectionIfNotExistsAsync(
+            UriFactory.CreateDatabaseUri(db),
+            new DocumentCollection { Id = collection });
+
+        runOnce = false;
+
+        log.Info("Initialization done!");
+    }
+
+    if (string.IsNullOrWhiteSpace(leaderboard) || string.IsNullOrWhiteSpace(playerId))
+        return req.CreateResponse(HttpStatusCode.BadRequest, "Both playerId and leaderboard should be provided in querystring, format is leaderboard/playerId");
+
+    try
+    {
+
+        var scores = client.CreateDocumentQuery<ScoreItem>(
+            UriFactory.CreateDocumentCollectionUri(db, collection), new FeedOptions { EnableCrossPartitionQuery = true });
+
+        // Get score for the current player by playerId        
+        var query =
+            (from s in scores
+             where s.PlayerId == playerId && s.Leaderboard == leaderboard
+             select s);
+
+        var player = query.ToList();
+        if (player.Count < 1)
+            return req.CreateResponse(HttpStatusCode.BadRequest, $"Player with Id = {playerId} in Leaderboard = {leaderboard} does not exist in the database");
+
+        if (player.Count > 1)
+            return req.CreateResponse(HttpStatusCode.BadRequest, $"There are more than one player with Id = {playerId} in Leaderboard = {leaderboard}. This is DB confuguration error - this combination should be unique");
+
+        var currentPlayer = player.First();
+
+        // Calculate global rank for the current player by playerId 
+        int globalRank;
+
+        query = from c in scores
+                where c.Score > currentPlayer.Score && c.Leaderboard == leaderboard
+                select c;
+
+        globalRank = query.Count<ScoreItem>() + 1;
+
+        List<LeaderboardItem> sortedResult = new List<LeaderboardItem>();
+
+        // Get and add to output specific number of players (radiusOfLeaderboard constant) with scores below the current player)
+        query =
+            (from s in scores
+             where s.Score < currentPlayer.Score && s.Leaderboard == leaderboard
+             orderby s.Score descending
+             select s).Take(radiusOfLeaderboard);
+
+        var sortedList = query.ToList<ScoreItem>().OrderBy(a => a.Score);
+
+        foreach (ScoreItem scr in sortedList)
+        {
+            sortedResult.Add(new LeaderboardItem { Rank = 0, Player = scr.Player, Score = scr.Score });
+        }
+        // Add current player to output
+        sortedResult.Add(new LeaderboardItem { Rank = globalRank, Player = currentPlayer.Player, Score = currentPlayer.Score });
+
+        // Get and add to output specific number of players (radiusOfLeaderboard constant) with scores above the current player)
+        query =
```

Comment:
This query removes all players with the same score as the current player, hence not returning a correct leaderboard result

Reply:
I thought that it doesn't make a sense to return players with the same score as requested playerId because we can't say they have higher or lower rank than current. So I implemented approach to show only those who has higher or lower scores (ranks). We need to decide how should we handle this situation so I can fix to implement this behavior.

Reply:
I think it's important to show people with the same rank (and same score) as the current player to fill out the around me result. So it might be that you get a result that only show a few that is ahead of you and many that have the same score as you.